### PR TITLE
Clarify what parameters are sent on an update

### LINKED ISSUE #350: Clarify expectations of a broker when parameters are updated
The spec has a gap when it comes to which parameters are sent to the broker during an update.  It's not clear whether _all_ the parameters should be resent, or only the ones that have changed.  We should clarify this so that broker authors have a correct expectation of what they will receive.

### PR DESCRIPTION
**Pull request by duglin**
Closes #350

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by angarg12** [comment]
LGTM

File: spec.md [hunk 1]
```diff
@@ -900,13 +900,28 @@ The following HTTP Headers are defined for this operation:
...
 | previous_values | object | Information about the service instance prior to the update. |
 | previous_values.service_id | string | Deprecated; determined to be unnecessary as the value is immutable. If present, it MUST be the ID of the service for the service instance. |
 | previous_values.plan_id | string | If present, it MUST be the ID of the plan prior to the update. |
 | previous_values.organization_id | string | Deprecated as it was redundant information. Organization for the service instance MUST be provided by platforms in the top-level field `context`. If present, it MUST be the ID of the organization specified for the service instance. |
 | previous_values.space_id | string | Deprecated as it was redundant information. Space for the service instance MUST be provided by platforms in the top-level field `context`. If present, it MUST be the ID of the space specified for the service instance. |
 
+Note: The `parameters` specified are expected to be the values specified
+by an end-user. Whether the user choose to include the complete set of
```
Comment (mattmcneeney) [inline spec.md]: *chooses

File: spec.md [hunk 2]
```diff
@@ -900,13 +900,28 @@ The following HTTP Headers are defined for this operation:
...
+allowable parameters or just a subset (or even none) is their choice. How a
```
Comment (mattmcneeney) [inline spec.md]: ~allowable parameters~ configuration options

File: spec.md [hunk 3]
```diff
@@ -900,13 +900,28 @@ The following HTTP Headers are defined for this operation:
...
+particular parameter) is out of scope of this specification - with the
+exception that if this field is not present in the request then the
+Service Broker MUST NOT change the parameters of the instance as a result of
+this request.
+
+Since some Service Instances will provide a `dashboard_url`, it is possible
+that a user has modified some of these parameters via the dashboard and
+therefore the Platform might not be aware of these changes. For this reason,
+Platforms SHOULD NOT include any parameters on the request that
+the user did not explicitly specify in their request for the update.
```
Comment (mattmcneeney) [inline spec.md]: This addition to the spec is very helpful IMO

File: spec.md (hunk 1 above)
Comment (duglin) [reply spec.md]: fixed

File: spec.md (hunk 2 above)
Comment (duglin) [reply spec.md]: fixed

**Comment by fmui** [comment]
LGTM

**Comment by duglin** [comment]
rebased

**Comment by mattmcneeney** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by duglin** [comment]
rebased

File: spec.md [hunk 4]
```diff
@@ -912,6 +912,21 @@ The following HTTP Headers are defined for this operation:
 | organization_id | string | Deprecated as it was redundant information. Organization for the Service Instance MUST be provided by Platforms in the top-level field `context`. If present, it MUST be the ID of the organization specified for the Service Instance. |
 | space_id | string | Deprecated as it was redundant information. Space for the Service Instance MUST be provided by Platforms in the top-level field `context`. If present, it MUST be the ID of the space specified for the Service Instance. |
 
+Note: The `parameters` specified are expected to be the values specified
+by an end-user. Whether the user chooses to include the complete set of
```
Comment (n3wscott) [inline spec.md]: >an end-user

Does this imply a person is filling these values out vs an automated system? Does end-user also include a Platform?

**Comment by n3wscott** [comment]
LGTM

File: spec.md (hunk 4 above)
Comment (mattmcneeney) [reply spec.md]: In CF, configuration parameters are chosen by a person rather than the system.

**Comment by mattmcneeney** [comment]
LGTM. Thanks @duglin

**Comment by angarg12** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by fmui** [comment]
LGTM

**Pull request merged**